includes/api/class-wc-admin-rest-reports-variations-controller.php

```diff
@@ -0,0 +1,296 @@
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @param Array $object Object data.
+	 * @return array        Links for the given post.
+	 */
+	protected function prepare_links( $object ) {
+		$links = array(
+			'product'   => array(
+				'href' => rest_url( sprintf( '/%s/%s/%d', $this->namespace, 'products', $object['product_id'] ) ),
+			),
+			'variation' => array(
+				'href' => rest_url( sprintf( '/%s/%s/%d/%s/%d', $this->namespace, 'products', $object['product_id'], 'variation', $object['variation_id'] ) ),
+			),
+		);
+
+		return $links;
+	}
+
+	/**
+	 * Get the Report's schema, conforming to JSON Schema.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'report_varitations',
+			'type'       => 'object',
+			'properties' => array(
+				'product_id'    => array(
+					'type'        => 'integer',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+					'description' => __( 'Product ID.', 'wc-admin' ),
+				),
+				'variation_id'  => array(
+					'type'        => 'integer',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+					'description' => __( 'Product ID.', 'wc-admin' ),
+				),
+				'items_sold'    => array(
+					'type'        => 'integer',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+					'description' => __( 'Number of items sold.', 'wc-admin' ),
+				),
+				'gross_revenue' => array(
+					'type'        => 'number',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+					'description' => __( 'Total gross revenue of all items sold.', 'wc-admin' ),
+				),
+				'orders_count'  => array(
+					'type'        => 'integer',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+					'description' => __( 'Number of orders product appeared in.', 'wc-admin' ),
+				),
+				'extended_info' => array(
+					'name'       => array(
+						'type'        => 'string',
+						'readonly'    => true,
+						'context'     => array( 'view', 'edit' ),
+						'description' => __( 'Product name.', 'wc-admin' ),
+					),
+					'price'      => array(
+						'type'        => 'integer',
+						'readonly'    => true,
+						'context'     => array( 'view', 'edit' ),
+						'description' => __( 'Product price.', 'wc-admin' ),
+					),
+					'image'      => array(
+						'type'        => 'string',
+						'readonly'    => true,
+						'context'     => array( 'view', 'edit' ),
+						'description' => __( 'Product image.', 'wc-admin' ),
+					),
+					'permalink'  => array(
+						'type'        => 'string',
+						'readonly'    => true,
+						'context'     => array( 'view', 'edit' ),
+						'description' => __( 'Product link.', 'wc-admin' ),
+					),
+					'attributes' => array(
+						'type'        => 'object',
```

Comment:
Better update this to an array, so we can do like in our endpoints on WooCommerce.

This is how we do for variations:

```php
'attributes'            => array(
	'description' => __( 'List of attributes.', 'woocommerce' ),
	'type'        => 'array',
	'context'     => array( 'view', 'edit' ),
	'items'       => array(
		'type'       => 'object',
		'properties' => array(
			'id'     => array(
				'description' => __( 'Attribute ID.', 'woocommerce' ),
				'type'        => 'integer',
				'context'     => array( 'view', 'edit' ),
			),
			'name'   => array(
				'description' => __( 'Attribute name.', 'woocommerce' ),
				'type'        => 'string',
				'context'     => array( 'view', 'edit' ),
			),
			'option' => array(
				'description' => __( 'Selected attribute term name.', 'woocommerce' ),
				'type'        => 'string',
				'context'     => array( 'view', 'edit' ),
			),
		),
	),
),
```

So the response should be changed to:

```json
{
    "attributes": [
        {
            "id": 1,
            "name": "Color",
            "option": "Green"
        }
    ]
}
```

Reply:
Great idea, thanks. Updated now and here is what a response item now looks like:

```js
    {
        "product_id": 12,
        "variation_id": 28,
        "items_sold": 2,
        "gross_revenue": 84,
        "orders_count": 1,
        "extended_info": {
            "name": "Hoodie",
            "price": 42,
            "image": "<img width=\"324\" height=\"324\" src=\"http://tangaroa.test/wp-content/uploads/2018/11/hoodie-2-324x324.jpg\" class=\"attachment-woocommerce_thumbnail size-woocommerce_thumbnail\" alt=\"\" srcset=\"http://tangaroa.test/wp-content/uploads/2018/11/hoodie-2-324x324.jpg 324w, http://tangaroa.test/wp-content/uploads/2018/11/hoodie-2-150x150.jpg 150w, http://tangaroa.test/wp-content/uploads/2018/11/hoodie-2-300x300.jpg 300w, http://tangaroa.test/wp-content/uploads/2018/11/hoodie-2-768x768.jpg 768w, http://tangaroa.test/wp-content/uploads/2018/11/hoodie-2-416x416.jpg 416w, http://tangaroa.test/wp-content/uploads/2018/11/hoodie-2-100x100.jpg 100w, http://tangaroa.test/wp-content/uploads/2018/11/hoodie-2.jpg 801w\" sizes=\"(max-width: 324px) 100vw, 324px\" />",
            "permalink": "http://tangaroa.test/product/hoodie/",
            "attributes": [
                {
                    "id": 1,
                    "name": "color",
                    "option": "Red"
                },
                {
                    "id": 0,
                    "name": "logo",
                    "option": "No"
                }
            ]
        },
        "_links": {
            "product": [
                {
                    "href": "http://tangaroa.test/wp-json/wc/v3/products/12"
                }
            ],
            "variation": [
                {
                    "href": "http://tangaroa.test/wp-json/wc/v3/products/12/variation/28"
                }
            ]
        }
    }
```